core/src/main/scala/org/apache/spark/status/protobuf/ResourceProfileWrapperSerializer.scala

```diff
@@ -18,19 +18,26 @@
 package org.apache.spark.status.protobuf
 
 import org.apache.spark.status.ResourceProfileWrapper
-import org.apache.spark.status.protobuf.ApplicationEnvironmentInfoWrapperSerializer.{deserializeResourceProfileInfo, serializeResourceProfileInfo}
 
-object ResourceProfileWrapperSerializer {
-  def serialize(input: ResourceProfileWrapper): Array[Byte] = {
+class ResourceProfileWrapperSerializer extends ProtoBufSerDe {
+
+  private val appEnvSerializer = new ApplicationEnvironmentInfoWrapperSerializer
```

Comment:
We can move the required method to `object ResourceProfileWrapperSerializer`  to avoid creating `appEnvSerializer`